app/bg/dbs/settings.js

```diff
@@ -39,7 +40,14 @@ export const setup = function (opts) {
     default_dat_ignore: '.git\n.dat\nnode_modules\n*.log\n**/.DS_Store\nThumbs.db\n',
     analytics_enabled: 1,
     dat_bandwidth_limit_up: 0,
-    dat_bandwidth_limit_down: 0
+    dat_bandwidth_limit_down: 0,
+    selected_search_engine: 0,
+    custom_search_engine_name: '',
+    custom_search_engine_url: '',
+    default_search_engines: [
+        {name:'DuckDuckGo',url:'https://www.duckduckgo.com/'},
+        {name:'Google',url:'https://www.google.com/search'}
+      ]
```

Comment:
I think we can modify these settings to be a little more robust. Rather than making the custom search engine separate, we can just have it be a part of the list. So:

```
selected_search_engine: 0,
search_engines: [
  {name: 'DuckDuckGo', url: 'https://www.duckduckgo.com/'},
  {name: 'Google', url: 'https://www.google.com/search'}
]
```

Then the user adds a custom engine by adding to that list. A few notes about this:

 1. The settings-db only saves strings and numbers, so we'll need to update the get/set to parse/stringify as JSON for the `search_engines` value. You can just hard code that as a conditional.
 2. Check your tab spacing there, the default_search_engines array is too indented.
 3. Also please put spaces after commas and colons in objects!

Reply:
Ahh, just store the whole JSON string. Right, that is way easier.